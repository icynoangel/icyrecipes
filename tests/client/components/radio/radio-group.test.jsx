import React from 'react';
import ReactDOM from 'react-dom';
import {mount} from 'enzyme';
import RadioGroup from './../../../../src/client/js/components/radio/radio-group';
import Radio from './../../../../src/client/js/components/radio/radio';
import renderer from 'react-test-renderer';

describe('<RadioGroup />', function() {
  this.getComponent = () => {
    return (
      <RadioGroup
        onChange={this.onChange}
        label={this.label}
        items={this.items}
      />
    );
  };

  this.items = [
    {
      label: 'Radio option 1',
      value: 'rad_1',
      checked: true
    },
    {
      label: 'Radio option 2',
      value: 'rad_2'
    },
    {
      label: 'Radio option 3',
      value: 'rad_3',
      disabled: true
    }
  ];

  beforeEach(() => {
    this.onChange = jest.fn();
    this.label = 'Radio Group';
  });

  it('Should render <RadioGroup />', () => {
    const radioGroup = renderer.create(this.getComponent()).toJSON();
    expect(radioGroup).toMatchSnapshot();
  });

  it('Should not call onChange for selected item', () => {
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Radio)
      .first()
      .simulate('click');
    expect(this.onChange).not.toHaveBeenCalled();
  });

  it('Should call onChange for unselected item', () => {
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Radio)
      .at(1)
      .simulate('click');
    expect(this.onChange).toHaveBeenCalledTimes(1);
    expect(this.onChange).toHaveBeenCalledWith(this.items[1].value);
  });

  it('Should not call onChange for disabled items', () => {
    const wrapper = mount(this.getComponent());
    wrapper
      .find(Radio)
      .at(2)
      .simulate('click');
    expect(this.onChange).not.toHaveBeenCalled();
  });
});
