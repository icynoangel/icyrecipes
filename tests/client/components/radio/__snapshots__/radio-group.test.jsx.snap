// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> Should render <RadioGroup /> 1`] = `
<div
  className="radio-group"
>
  <div
    className="radio-group__label"
  >
    Radio Group
  </div>
  <div
    className="radio-group__content"
  >
    <div
      className="radio --checked"
      onClick={[Function]}
    >
      <div
        className="radio__button"
      >
        <div
          className="radio__button__checked"
        />
      </div>
      <div
        className="radio__label"
      >
        Radio option 1
      </div>
    </div>
    <div
      className="radio"
      onClick={[Function]}
    >
      <div
        className="radio__button"
      >
        <div
          className="radio__button__checked"
        />
      </div>
      <div
        className="radio__label"
      >
        Radio option 2
      </div>
    </div>
    <div
      className="radio --disabled"
      onClick={[Function]}
    >
      <div
        className="radio__button"
      >
        <div
          className="radio__button__checked"
        />
      </div>
      <div
        className="radio__label"
      >
        Radio option 3
      </div>
    </div>
  </div>
</div>
`;
